client/infinispan-key-value-store-hotrod/src/main/java/org/infinispan/api/collections/reactive/client/impl/ContinuousQueryPublisherImpl.java

```diff
@@ -0,0 +1,152 @@
+package org.infinispan.api.collections.reactive.client.impl;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Flow;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.api.collections.reactive.CreatedKeyValueEntry;
+import org.infinispan.api.collections.reactive.KeyValueEntry;
+import org.infinispan.api.collections.reactive.RemovedKeyValueEntry;
+import org.infinispan.api.collections.reactive.UpdatedKeyValueEntry;
+import org.infinispan.api.search.reactive.ContinuousQueryPublisher;
+import org.infinispan.api.search.reactive.QueryParameters;
+import org.infinispan.query.api.continuous.ContinuousQuery;
+import org.infinispan.query.api.continuous.ContinuousQueryListener;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+import org.reactivestreams.FlowAdapters;
+import org.reactivestreams.Subscriber;
+
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.FlowableEmitter;
+
+/**
+ * Implements the {@link ContinuousQueryPublisher} using RXJava and Flowable API.
+ *
+ * @author Katia Aresti
+ * @since 10.0
+ */
+public class ContinuousQueryPublisherImpl<K, V> implements ContinuousQueryPublisher<K, V> {
```

Comment:
The query definition methods look like operators on a stream but they are not. They apply to this unique publisher instance and only before subscription.

I believe you should rework the API to separate the query builder methods from the publisher implementation (see my comment in the interface file).

Reply:
ok. So It's confusing ...

Reply:
re-reviewing this approach after looking to the web client